src/extension.ts

```diff
@@ -0,0 +1,48 @@
+import * as vscode from 'vscode';
+
+export function activate(context: vscode.ExtensionContext) {
+	
+	// Use the console to output diagnostic information (console.log) and errors (console.error)
+	// This line of code will only be executed once when your extension is activated
+	console.log('Congratulations, your extension "SM213" is now active!');
+
+	// The command has been defined in the package.json file
+	// Now provide the implementation of the command with registerCommand
+	// The commandId parameter must match the command field in package.json
+	let disposable = vscode.commands.registerCommand('sm213-assembly-syntax-highlighting.credit', () => {
+		// The code you place here will be executed every time your command is executed
+		// Display a message box to the user
+		vscode.window.showInformationMessage('0000 - 2021 Humanity!');
+	});
+
+	context.subscriptions.push(disposable);
+	context.subscriptions.push(HOVER_DEFN);
+}
+
+// Hovering
+
+const HOVER_DEFN = vscode.languages.registerHoverProvider({ scheme: 'file', language: 'asm' }, {
+	provideHover(document, position, token) {
+		const range = document.getWordRangeAtPosition(position);
+        	const word = document.getText(range);
+		// const hoverMsg: vscode.MarkdownString = new vscode.MarkdownString().appendText('GTFO!');
+		// hoverMsg.isTrusted = true;
+		if (word in DEFN) {return new vscode.Hover(DEFN[word]);}
+		// return new vscode.Hover(hoverMsg);
+	}
+});
+
+const DEFN: {[key: string]: vscode.MarkdownString} = {
+	// Format:
+	halt : new vscode.MarkdownString()
+		.appendCodeblock('halt', 'asm')
+		.appendText(`\n`)
+		.appendMarkdown('(stop execution)'),
+	nop : new vscode.MarkdownString()
+		.appendCodeblock('nop', 'asm')
+		.appendText(`\n`)
+		.appendMarkdown('(stop execution)'),
+
+};
```

Comment:
This is so cool! I guess we would just add more key value pairs to `DEFN` as necessary to support more hovered words?

Reply:
Thank you for your approval! The only problem being that I have I have difficulty getting vscode.MarkdownString() to work with tex, mathML and embed images. So in the worst case might need to drop the idea of displaying semantic but rather description in common English & code block.

Reply:
It's my first time too! We're all figuring this out together :))

I think you're right, the only thing `vscode.Hover(...)` can take as parameters are `vscode.MarkdownString`s which don't look like they support tex or mathML judging from [the documentation](https://code.visualstudio.com/api/references/vscode-api#MarkdownString) I can find. Markdown does support images though. If I add a `.appendMarkdown("![](https://i.ibb.co/DCDP5Jm/Screen-Shot-2021-09-28-at-1-08-56-PM.png)")`, I see this:

<img width="321" alt="Screen Shot 2021-09-29 at 12 22 17 AM" src="https://user-images.githubusercontent.com/63937200/135221939-4d6b392b-5dab-4bfe-936f-d5190a156742.png">

For some reason, I can only get this to work when the markdown image comes from an external link, it's not working for me when I try to give it the path to some local file (something like `"![](../assets/icon.png)"`).

This might be worth exploring in the future, but I don't think it's critical to the rest of the hovering functionality and can probably be looked into later/ in a different PR?